src/mongo/dbtests/query_stage_distinct.cpp

```diff
@@ -236,6 +236,73 @@ class QueryStageDistinctMultiKey : public DistinctBase {
     }
 };
 
+class QueryStageDistinctCompoundIndex : public DistinctBase {
+public:
+    void run() {
+        // insert documents with a: 1 and b: 1
+        for (size_t i = 0; i < 1000; ++i) {
+            insert(BSON("a" << 1 << "b" << 1));
+        }
+        // insert documents with a: 1 and b: 2
+        for (size_t i = 0; i < 1000; ++i) {
+            insert(BSON("a" << 1 << "b" << 2));
+        }
+        // insert documents with a: 2 and b: 1
+        for (size_t i = 0; i < 1000; ++i) {
+            insert(BSON("a" << 2 << "b" << 1));
+        }
+        // insert documents with a: 2 and b: 3
+        for (size_t i = 0; i < 1000; ++i) {
+            insert(BSON("a" << 2 << "b" << 3));
+        }
+
+        addIndex(BSON("a" << 1 << "b" << 1));
+
+        AutoGetCollectionForRead ctx(&_txn, ns());
+        Collection* coll = ctx.getCollection();
+
+        std::vector<IndexDescriptor*> indices;
+        coll->getIndexCatalog()->findIndexesByKeyPattern(&_txn, BSON("a" << 1 << "b" << 1), false, &indices);
+        verify(indices.size() == 1);
```

Comment:
Since this is a test, use ASSERT_EQ(indices.size(), 1), rather than verify(). Same for line 270, and elsewhere in this test.